### PR TITLE
CHET-273: Migration stack bugfixes; DB transfer bugfixes

### PR DESCRIPTION
**Pull request by bpar476**
These bugs were found while testing the migration end-to-end via the UI.

* State transitions during the DB migration were incorrect and duplicated
* Migration stack had some syntax errors

### CODE REVIEW

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java [hunk 1]
```diff
@@ -85,12 +85,8 @@ public FileSystemMigrationErrorReport performMigration() throws DatabaseMigratio
 
         Path pathToDatabaseFile = databaseArchivalService.archiveDatabase(tempDirectory, stageTransitionCallback);
 
-        migrationService.transition(MigrationStage.DB_MIGRATION_EXPORT_WAIT);
```
Comment (nanux) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java:88]: These transitions are done via `transitionCallbacks`?

File: templates/migration-stack/migration-helper.yml [hunk 2]
```diff
@@ -152,7 +152,7 @@ Resources:
                       exit 1
                     fi
                     aws s3 sync s3://${MigrationBucket}/db.dump/ $DATABASE_DOWNLOAD_DIR --region ${AWS::Region} | tee $DB_DUMP_LOG_FILE
-                    if [ ${PIPESTATUS[0]} != 0]; then
+                    if [ ${pipestatus} != 0]; then
```
Comment (nanux) [inline templates/migration-stack/migration-helper.yml:155]: 🤦 sorry didn't realise the `!Sub` will eagirly try to substitute it.

File: core/src/test/java/com/atlassian/migration/datacenter/core/aws/infrastructure/AWSMigrationHelperDeploymentServiceTest.java [hunk 3]
```diff
@@ -108,7 +108,7 @@ void shouldNameMigrationStackAfterApplicationStackWithSuffixAndStoreInContext()
     void shouldProvisionCloudformationStack() {
         givenMigrationStackHasStartedDeploying();
 
-        verify(mockCfn).provisionStack("https://trebuchet-aws-resources.s3.amazonaws.com/migration-helper.yml", DEPLOYMENT_ID, Collections.emptyMap());
+        verify(mockCfn).provisionStack("https://trebuchet-public-resources.s3.amazonaws.com/migration-helper.yml", DEPLOYMENT_ID, Collections.emptyMap());
```
Comment (nanux) [inline core/src/test/java/com/atlassian/migration/datacenter/core/aws/infrastructure/AWSMigrationHelperDeploymentServiceTest.java:111]: I assume you've uploaded the fixed migraiton helper.

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java (hunk 1 above)
Comment (bpar476) [reply core/src/main/java/com/atlassian/migration/datacenter/core/aws/db/DatabaseMigrationService.java:88]: Yes 👍

File: templates/migration-stack/migration-helper.yml (hunk 2 above)
Comment (bpar476) [reply templates/migration-stack/migration-helper.yml:155]: No worries, it's kinda silly.

File: core/src/test/java/com/atlassian/migration/datacenter/core/aws/infrastructure/AWSMigrationHelperDeploymentServiceTest.java (hunk 3 above)
Comment (bpar476) [reply core/src/test/java/com/atlassian/migration/datacenter/core/aws/infrastructure/AWSMigrationHelperDeploymentServiceTest.java:111]: Correct :D

**Pull request merged**